added some encapsulation

maciejDun opened the pull request:
Please Karl Bill check my code!

maciejDun committed:
added some encapsulation

karolk92 commented on src/main/java/com/example/battleship/BattleshipApplication.java:
```diff
@@ -46,8 +46,6 @@ public static void main(String[] args) {
 
```
Usun zakomentowane linie zeby nie zapuszczac kodu. Takie rzeczy nie powinny przechodzic w commitach. Zostaw jedynie zakomentowana linie ze SpringBootApplication - bedzie niezbedne do uruchomienia aplikacji.

karolk92 commented on src/main/java/com/example/battleship/game/Board.java:
```diff
@@ -25,3 +25,4 @@ public boolean checkAllSunkenShips(){
         return true;
```
linia 20 - zastanow sie nad nazwa metody, osoba widzaca nazwe metody powinna wiedziec co wlasciwie metoda robi. `areShipsSunken()`? moze wymyslisz cos lepszego

karolk92 commented on src/main/java/com/example/battleship/game/BoardService.java:
```diff
@@ -121,31 +121,6 @@ public static Board initPlayer2Board(){
-//                fourDeckerShip.getShipField(3), fourDeckerShip));
-//        System.out.println();
-//        return board;
     }
```
Zamiast printowac informacje poprzes `System.out.print` mozesz dac nad klasa `@Slf4j` a w kodzie bedziesz mogl uzywac loggera np `log.info("Board initialized");` To najprostszy sposob na logowanie a aplikacjach springbootowych

karolk92 commented on src/main/java/com/example/battleship/game/Field.java:
```diff
@@ -27,3 +27,4 @@ public int hashCode() {
         return this.position.hashCode() * 31;
```
linia 6 - pole final, zeby zapobiec modyfikacji, poczytaj o obiektach immatable i jak je tworzyc

karolk92 replied:
Mozesz tez poczytac o Lomboku, to taki fajny mechanizm do generowania np. getterow przez @Getter itd, pozbywasz sie tzw boilerplate code'u

karolk92 commented on src/main/java/com/example/battleship/game/FourDeckerShip.java:
```diff
@@ -8,9 +8,5 @@ public FourDeckerShip(Map<Field, ShipPart> shipParts) {
         super(ShipType.FOUR_DECKER, shipParts);
```
fajnie ze pomyslales o dziedziczeniu

karolk92 commented on src/main/java/com/example/battleship/game/Game.java:
```diff
@@ -55,7 +55,7 @@ public boolean shoot(String position, String playerName){
         }
```
linia 17 - enter, pilnuj odstepow pomiedzy metodami, poczytaj tez o podstawowych zasadach clean code'u - w jakiej kolejnosci sytuowac metody.

karolk92 replied:
metoda shoot  jest za dluga - podziel na mniejsze kawalki - wyglada jakbys mogl wyciagnac wspolny kawalek (linie 33-42 i 34-53), ale nie jestem pewien - sprobuj, a jesli sie nie uda to po prostu podziel na mniejsze kawalki

karolk92 commented on src/main/java/com/example/battleship/game/Ship.java:
```diff
@@ -29,13 +29,6 @@ public boolean checkIfSunken(){
-    public Map<Field, ShipPart> getShipParts() {
-        return shipParts;
-    }
-    abstract Field getShipField(int i);
```
a moze tutaj bys zaimplementowal logike ktora robisz poziom nizej? `getShipParts(int i) { return shipParts.get(i).getPosition()` zamiast implementowac to w kazdym dziecku?

maciejDun replied:
this has already been changed, there is no need for getting shipParts from ship childs, nor in Ship mother

karolk92 commented on src/main/java/com/example/battleship/game/Player.java:
```diff
@@ -50,3 +50,4 @@ public boolean getIsLost() {
         return isLost;
```
logger o ktorym pisalem zamiast printowania

karolk92 commented on src/main/java/com/example/battleship/game/Game.java:
```diff
@@ -78,7 +78,7 @@ private boolean validateName(String name){
+    private void checkAllSunkenShips(Player player){
         if (player.checkAllSunkenShips()){
             System.out.println("Player: " + player.getUsername() + " have no ships left on his board!!!!");
             player.setIsLost();
```
cos mi to `setIsLost()` nie pasuje, zmienilbym nazwe

karolk92 reviewed (changes requested):
Mamy pole do dalszej nauki - rozwaz w jaki sposob zmienic ture i zapobiec strzeleniu 2x przez tego samego uzytkownika jesli spudlowal. Poczyn tez pierwsze proby rzucania wyjatkow - przykladowo jesli to nie jest tura gracza ktory probuje strzelic, albo gra juz sie skonczyla, albo jeszcze nie zaczela a gracz juz chce strzelac etc.

maciejDun committed:
-in Board class added @Getter, new name of method areAllShipsSunken for more readability, also added final keyword to class and map
-BoardService: added @Slf4j and logs for printing messages
-Field class is unmutable
-Ship: added final to shitType and Map of shipParts
